src/shared.ts

```diff
@@ -31,3 +31,130 @@ export function implementsCmp<T>(x: T | Cmp<T>): x is Cmp<T> {
         x.cmp.length === 1
     );
 }
+
+export function defaultEqualityFn<K>(a: K, b: K): boolean {
+    if (implementsEq(a)) {
+        return a.eq(b);
+    }
+    return a === b;
+}
+
+export function defaultComparatorFn<K>(a: K, b: K): Ordering {
+    if (implementsCmp(a)) {
+        return a.cmp(b);
+    }
+    if (a < b) return Ordering.Less;
+    if (a > b) return Ordering.Greater;
+    return Ordering.Equal;
+}
+
+export interface ReadonlySetLike<T> {
+    [Symbol.iterator](): Iterator<T>;
+    size: number;
+    entries(): IterableIterator<[T, T]>;
+    forEach(fn: (value: T) => void): void;
+    has(value: T): boolean;
+    keys(): IterableIterator<T>;
+    values(): IterableIterator<T>;
+}
+
+export interface SetLike<T> extends ReadonlySetLike<T> {
+    add(value: T): this;
+    delete(value: T): boolean;
+    clear(): void;
+}
+
+export interface SetUtils<T> {
+    difference<U>(other: ReadonlySetLike<U>): Set<T>;
+    intersection(other: ReadonlySetLike<T>): Set<T>;
+    isDisjointFrom(other: ReadonlySetLike<T>): boolean;
+    isSubsetOf(other: ReadonlySetLike<T>): boolean;
+    isSupersetOf(other: ReadonlySetLike<T>): boolean;
+    symmetricDifference<U>(other: ReadonlySetLike<U>): Set<T | U>;
+    union<U>(other: ReadonlySetLike<U>): Set<T | U>;
+}
```

Comment:
why do you even need to define this stuff when it's already defined in the spec? a `Set` has this

Did you add this because TS doesn't define a ` [Symbol.iterator](): Iterator<T>;`?

![image](https://github.com/trvswgnr/collectionmaxxing/assets/27996712/c5e95262-3dc2-4755-9f19-6d99a1267b26)

Reply:
this is wild lol i did **not** know there were type definitions for this.. i don't have them on my machine? tsconfig has "esnext" in the lib.